guillotina/contrib/dbusers/users.py

```diff
@@ -28,4 +30,16 @@ class DBUserIdentifier:
             # User is disabled
             return
 
+        if not hasattr(user, "_groups_cache"):
+            user._groups_cache = {}
+
+        # load groups
+        for ident in user.groups:
+            try:
+                group = await navigate_to(container, f"groups/{ident}")
+            except KeyError:
+                continue
+
+            user._groups_cache[ident] = GuillotinaGroup(ident, roles=group.roles)
```

Comment:
This is so wired to be part of a public API, no? At least, what we should answer to the question how you can provide your own auth API?

Reply:
Yes, I agree. This is something I want to discuss why @bloodbare . The point of this change is not breaking compatibility with existing custom implementations, but I think this should go in a dedicated method like `async get_groups(user)` in the user_identifier.

Reply:
We can't loose the main focus of the issue, we are touching this part, because we want to clarify the "extensible" API to roll our own auth/identification system. and on the way we found the this API it's not complete (ready) on the dbusers referential implementation.

Reply:
I updated `IPrincipal` to make explicit the field `_groups_cache`. Also, we don't break the existing implementation of IGroups that was using the `_groups_cache`.